packages/beacon-node/src/execution/builder/http.ts

```diff
@@ -34,10 +35,13 @@ export class ExecutionBuilderHttp implements IExecutionBuilder {
   faultInspectionWindow: number;
   allowedFaults: number;
 
-  constructor(opts: ExecutionBuilderHttpOpts, config: ChainForkConfig) {
+  constructor(opts: ExecutionBuilderHttpOpts, config: ChainForkConfig, metrics: Metrics | null = null) {
     const baseUrl = opts.urls[0];
     if (!baseUrl) throw Error("No Url provided for executionBuilder");
-    this.api = getClient({baseUrl, timeoutMs: opts.timeout}, {config});
+    this.api = getClient(
+      {baseUrl, timeoutMs: opts.timeout},
+      {config, metrics: metrics?.builderHttpClient as ApiMetrics | undefined}
```

Comment:
need to cast because there are different `CollectFn` types defined in different packages